feature-libs/asm/components/services/asm-component.service.ts

```diff
@@ -46,6 +51,7 @@ export class AsmComponentService {
     protected winRef: WindowRef,
     // TODO(CXSPA-3090): Remove optional flag in 7.0 where service is used
     @Optional() protected asmEnablerService?: AsmEnablerService,
+    @Optional() protected routingService?: RoutingService,
     @Optional() protected asmDeepLinkService?: AsmDeepLinkService
```

Comment:
[MAJOR REMARK - breaking change]
We should add new constructor parameters only as the _last_ parameters, being optional.
In the case above the new parameter is optional, but it's added in the place of the `asmDeepLinkService`, which is a breaking change.

Why it's breaking change:
A compiling error will happen for a customer that used to extend this service and calling it's `super` constructor, and then they upgrade Spartacus to the v6.5:
```ts
export class MyAsmComponentService extends AsmComponentService {

  constructor(
    protected authService: AuthService,
    protected csAgentAuthService: CsAgentAuthService,
    protected winRef: WindowRef,
    protected asmEnablerService?: AsmEnablerService,
    protected asmDeepLinkService?: AsmDeepLinkService
  ){
     super(
       authService,
       csAgentAuthService,
       winRef,
       asmEnablerService,
       asmDeepLinkService // <---- error, the new version of this constructor expects `routingService` in this place!
  }
}

Reply:
Thank you. It seems that this occurred after we merged the 6.5.x branch into our own. we will move routingService to the end.

Reply:
I used `inject`
` protected routingService = inject(RoutingService);`